### ISSUE TITLE
 modalRef.componentInstance problem

### ISSUE THREAD

**Issue by BehnamAbdy** [ng-bootstrap/ng-bootstrap#948]
i open modal in app with component instead of template then i need to pass the object model to modal component, the problem is that the typescript gives error of  modalRef.componentInstance not exists as property, i exactly copy the sample form demo page but again give the same error and never get the @input variable populated on modal content class,

this is error
Cannot set property 'model' of undefined

@Component({
    selector: 'company-list',
    templateUrl: './app/components/company/company-list.component.html'
})
export class CompanyListComponent implements {
    private modalRes: Company;

    constructor(private modalService: NgbModal) {
    }

    open(company: Company) {
        const modalRef = this.modalService.open(CompanyAddComponent);
        modalRef.componentInstance.name = 'some name';  //** this line gives error

        modalRef.result.then((result) => {           
        }, (reason) => {
           
        });     
    }

    createCompany(model: Company) {
        this.companyService.createCompany(model);
    }
}

and inside the modal i declare this varable to get the passed in value
  @Input() model: Company;
but it is always null

**Comment by jnizet** [comment]
Without a plunkr, and at least the exact and complete error you get, it's very hard to know what the problem can be. 

The only thing I can notice is that you expect the field `model`, of type `Company`, to be not null, but the only attribute you set in your code is the attribute `name`, to the value `'some name'`.

You should ask this question on Stackoverflow, with a complete minimal example reproducing the problem, and the exact and complete error message you get.

**Comment by pkozlowski-opensource** [comment]
As @jnizet have said we can't help more without a minimal reproduce scenario in plunker. See the full reasoning here: https://github.com/ng-bootstrap/ng-bootstrap/blob/master/README.md#you-think-youve-found-a-bug

Closing as no reproduce scenario was given, but happy to re-open when you can provide one.

**Comment by BehnamAbdy** [comment]
Actually     modalRef.componentInstance is null even when I copy the exact code from  modal sample in plunker https://ng-bootstrap.github.io/app/components/modal/demos/component/plnkr.html
Maybe I've installed a n old version.I use '^1.0.0-alpha.9'

**Comment by pkozlowski-opensource** [comment]
> Maybe I've installed a n old version

Most probably. Or you've got an old version cached somewhere. Here is a plunker with version forced as 1.0.0-alpha.9 and it works perfectly fine: http://plnkr.co/edit/JRcdgBCeRsH1V9FgjdhZ?p=preview

tl;dr; It looks like a version mismatch / caching pb on your end.

**Comment by franzipyweb** [comment]
I also got the same error with the modalRef.componentInstance when I use .result.then in the modal itself.

`const modalRef = this.modalService.open(
      AdminDepositViewMoreModalComponent,
      { size: 'lg' }).result.then((result) => {
        if(result) {
          setTimeout(() => {
            this.buildTable()
          })
        }
      }).catch(function(reason) {
        console.log(reason)
      });

    modalRef.componentInstance.deposit = deposit`
